test/integration/xds_config_tracker_integration_test.cc

```diff
@@ -0,0 +1,291 @@
+    });
+
+    // Build and add the xDS cluster config.
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      auto* xds_cluster = bootstrap.mutable_static_resources()->add_clusters();
+      xds_cluster->MergeFrom(ConfigHelper::buildStaticCluster(
+          std::string(XDS_CLUSTER_NAME),
+          /*port=*/0, ipVersion() == Network::Address::IpVersion::v4 ? "127.0.0.1" : "::1"));
+      ConfigHelper::setHttp2(*xds_cluster);
+    });
+
+    // Add static runtime values.
+    config_helper_.addRuntimeOverride("whatevs", "yar");
+
+    // Set up the RTDS runtime layer.
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      auto* layer = bootstrap.mutable_layered_runtime()->add_layers();
+      layer->set_name("some_rtds_layer");
+      auto* rtds_layer = layer->mutable_rtds_layer();
+      rtds_layer->set_name("some_rtds_layer");
+      auto* rtds_config = rtds_layer->mutable_rtds_config();
+      rtds_config->set_resource_api_version(envoy::config::core::v3::ApiVersion::V3);
+      auto* api_config_source = rtds_config->mutable_api_config_source();
+      api_config_source->set_transport_api_version(envoy::config::core::v3::ApiVersion::V3);
+      api_config_source->set_api_type(this->sotwOrDelta() == Grpc::SotwOrDelta::Sotw
+                                          ? envoy::config::core::v3::ApiConfigSource::GRPC
+                                          : envoy::config::core::v3::ApiConfigSource::DELTA_GRPC);
+      api_config_source->set_set_node_on_first_message_only(true);
+      api_config_source->add_grpc_services()->mutable_envoy_grpc()->set_cluster_name(
+          XDS_CLUSTER_NAME);
+    });
+
+    // Add test xDS config tracer.
+    config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      auto* tracer_extension = bootstrap.mutable_xds_config_tracker_extension();
+      tracer_extension->set_name("envoy.config.xds.test_xds_tracer");
+      tracer_extension->mutable_typed_config()->PackFrom(
+          test::envoy::config::xds::TestXdsConfigTracker());
+    });
+  }
+
+  void TearDown() override {
+    if (xds_connection_ != nullptr) {
+      cleanUpXdsConnection();
+    }
+  }
+
+  void initialize() override {
+    // The tests infra expects the xDS server to be the second fake upstream, so
+    // we need a dummy data plane cluster.
+    setUpstreamCount(1);
+    setUpstreamProtocol(Http::CodecType::HTTP2);
+    HttpIntegrationTest::initialize();
+    registerTestServerPorts({});
+    initial_load_success_ = test_server_->counter("runtime.load_success")->value();
+  }
+
+  void acceptXdsConnection() {
+    createXdsConnection();
+    AssertionResult result = xds_connection_->waitForNewStream(*dispatcher_, xds_stream_);
+    RELEASE_ASSERT(result, result.message());
+    xds_stream_->startGrpcStream();
+  }
+
+  std::string getRuntimeKey(const std::string& key) {
```

Comment:
Why is the runtime and the admin needed for this test?
Can you use something that doesn't require it, maybe sending back a static cluster?

Reply:
Done. the test is using RTDS to call the extension, and removed the admin.